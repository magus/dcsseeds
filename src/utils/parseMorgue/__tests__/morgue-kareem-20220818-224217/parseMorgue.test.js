import { test_parseMorgue } from '../test_parseMorgue';

// incorrectly parsed 0.30-a0 as non-trunk
const morgue_url = 'https://crawl.kelbi.org/crawl/morgue/kareem/morgue-kareem-20220818-224217.txt.gz';

test(morgue_url, async () => {
  const result = await test_parseMorgue(morgue_url);
  expect(result).toMatchSnapshot();
});
