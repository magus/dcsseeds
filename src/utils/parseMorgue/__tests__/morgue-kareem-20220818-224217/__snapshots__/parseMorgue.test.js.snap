// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`https://crawl.kelbi.org/crawl/morgue/kareem/morgue-kareem-20220818-224217.txt.gz 1`] = `
{
  "background": "Gladiator",
  "datetime": 2022-08-18T22:42:17.000Z,
  "eventCount": 118,
  "eventErrors": [],
  "events": [
    {
      "branch": "Dungeon",
      "level": "1",
      "location": "Dungeon:1",
      "note": "kareem the Merfolk Gladiator began the quest for the Orb.",
      "turn": "0",
      "type": "first-event",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "16",
        "level": "1",
        "mp": "1",
      },
      "level": "1",
      "location": "Dungeon:1",
      "note": "Reached XP level 1. HP: 16/16 MP: 1/1",
      "turn": "0",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "23",
        "level": "2",
        "mp": "2",
      },
      "level": "1",
      "location": "Dungeon:1",
      "note": "Reached XP level 2. HP: 23/23 MP: 2/2",
      "turn": "254",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "god": "Unknown",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Found a faded altar of an unknown god.",
      "turn": "1170",
      "type": "altar",
    },
    {
      "branch": "Dungeon",
      "data": {
        "name": "Mowitoev's Assorted Antiques",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Found Mowitoev's Assorted Antiques.",
      "turn": "1212",
      "type": "shop",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "28",
        "level": "3",
        "mp": "3",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Reached XP level 3. HP: 28/28 MP: 3/3",
      "turn": "1218",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "5",
        "skill": "Dodging",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Reached skill level 5 in Dodging",
      "turn": "1476",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Natasha",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Noticed Natasha",
      "turn": "1723",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Natasha",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Killed Natasha",
      "turn": "1730",
      "type": "unique-killed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "35",
        "level": "4",
        "mp": "4",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Reached XP level 4. HP: 34/35 MP: 4/4",
      "turn": "1731",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Natasha",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Noticed Natasha",
      "turn": "1908",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Natasha",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Killed Natasha",
      "turn": "1926",
      "type": "unique-killed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Crazy Yiuf",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Noticed Crazy Yiuf",
      "turn": "2165",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Natasha",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Noticed Natasha",
      "turn": "2250",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Natasha",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Killed Natasha",
      "turn": "2263",
      "type": "unique-killed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "6",
        "skill": "Dodging",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Reached skill level 6 in Dodging",
      "turn": "2264",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "41",
        "level": "5",
        "mp": "5",
      },
      "level": "2",
      "location": "Dungeon:2",
      "note": "Reached XP level 5. HP: 18/41 MP: 5/5",
      "turn": "2264",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "name": "Herouq's Antique Armour Boutique",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Found Herouq's Antique Armour Boutique.",
      "turn": "2930",
      "type": "shop",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Grinder",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Noticed Grinder",
      "turn": "2997",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Sigmund",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Noticed Sigmund",
      "turn": "3077",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "name": "Lasoaju's General Store",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Found Lasoaju's General Store.",
      "turn": "3077",
      "type": "shop",
    },
    {
      "branch": "Dungeon",
      "data": {
        "name": "Syatay's General Store",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Found Syatay's General Store.",
      "turn": "3077",
      "type": "shop",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Jessica",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Noticed Jessica",
      "turn": "3378",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Jessica",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Killed Jessica",
      "turn": "3386",
      "type": "unique-killed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "48",
        "level": "6",
        "mp": "6",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Reached XP level 6. HP: 40/48 MP: 6/6",
      "turn": "3442",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Sigmund",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Killed Sigmund",
      "turn": "3572",
      "type": "unique-killed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "5",
        "skill": "Long Blades",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Reached skill level 5 in Long Blades",
      "turn": "3573",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "7",
        "skill": "Dodging",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Reached skill level 7 in Dodging",
      "turn": "3573",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Grinder",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Killed Grinder",
      "turn": "3916",
      "type": "unique-killed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "8",
        "skill": "Dodging",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Reached skill level 8 in Dodging",
      "turn": "3917",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "55",
        "level": "7",
        "mp": "7",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Reached XP level 7. HP: 55/55 MP: 7/7",
      "turn": "3917",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "name": "Kenoem's Magic Scroll Boutique",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Found Kenoem's Magic Scroll Boutique.",
      "turn": "3965",
      "type": "shop",
    },
    {
      "branch": "Ossuary",
      "data": {
        "level": "5",
        "skill": "Fighting",
      },
      "location": "Ossuary",
      "note": "Reached skill level 5 in Fighting",
      "turn": "4598",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "name": "Dimas's Elemental Accessories",
      },
      "level": "4",
      "location": "Dungeon:4",
      "note": "Found Dimas's Elemental Accessories.",
      "turn": "5491",
      "type": "shop",
    },
    {
      "branch": "Dungeon",
      "data": {
        "gold": "330",
        "is_valid_shop": true,
        "item": "lightning rod (4/4)",
      },
      "level": "4",
      "location": "Dungeon:4",
      "note": "Bought a lightning rod (4/4) for 330 gold pieces",
      "turn": "5498",
      "type": "bought",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "9",
        "skill": "Dodging",
      },
      "level": "5",
      "location": "Dungeon:5",
      "note": "Reached skill level 9 in Dodging",
      "turn": "6065",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "62",
        "level": "8",
        "mp": "8",
      },
      "level": "5",
      "location": "Dungeon:5",
      "note": "Reached XP level 8. HP: 33/62 MP: 8/8",
      "turn": "6065",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "1",
        "skill": "Evocations",
      },
      "level": "5",
      "location": "Dungeon:5",
      "note": "Reached skill level 1 in Evocations",
      "turn": "6632",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Duvessa",
      },
      "level": "5",
      "location": "Dungeon:5",
      "note": "Noticed Duvessa",
      "turn": "6751",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Duvessa",
      },
      "level": "5",
      "location": "Dungeon:5",
      "note": "Killed Duvessa",
      "turn": "6791",
      "type": "unique-killed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Dowan",
      },
      "level": "5",
      "location": "Dungeon:5",
      "note": "Noticed Dowan",
      "turn": "6990",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Dowan",
      },
      "level": "5",
      "location": "Dungeon:5",
      "note": "Killed Dowan",
      "turn": "7007",
      "type": "unique-killed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "10",
        "skill": "Dodging",
      },
      "level": "5",
      "location": "Dungeon:5",
      "note": "Reached skill level 10 in Dodging",
      "turn": "7306",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "69",
        "level": "9",
        "mp": "9",
      },
      "level": "6",
      "location": "Dungeon:6",
      "note": "Reached XP level 9. HP: 69/69 MP: 9/9",
      "turn": "8231",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "11",
        "skill": "Dodging",
      },
      "level": "6",
      "location": "Dungeon:6",
      "note": "Reached skill level 11 in Dodging",
      "turn": "8248",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "12",
        "skill": "Dodging",
      },
      "level": "7",
      "location": "Dungeon:7",
      "note": "Reached skill level 12 in Dodging",
      "turn": "9228",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "13",
        "skill": "Dodging",
      },
      "level": "7",
      "location": "Dungeon:7",
      "note": "Reached skill level 13 in Dodging",
      "turn": "9619",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "5",
        "skill": "Evocations",
      },
      "level": "7",
      "location": "Dungeon:7",
      "note": "Reached skill level 5 in Evocations",
      "turn": "9619",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "god": "Okawaru",
      },
      "level": "7",
      "location": "Dungeon:7",
      "note": "Found an iron altar of Okawaru.",
      "turn": "9620",
      "type": "altar",
    },
    {
      "branch": "Temple",
      "data": {
        "god": "Wu Jian",
      },
      "location": "Temple",
      "note": "Became a worshipper of the Wu Jian Council",
      "turn": "9837",
      "type": "join-god",
    },
    {
      "branch": "Dungeon",
      "data": {
        "name": "Atomniv's Armour Shop",
      },
      "level": "7",
      "location": "Dungeon:7",
      "note": "Found Atomniv's Armour Shop.",
      "turn": "9911",
      "type": "shop",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Grum",
      },
      "level": "8",
      "location": "Dungeon:8",
      "note": "Noticed Grum",
      "turn": "10077",
      "type": "unique-noticed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "who": "Grum",
      },
      "level": "8",
      "location": "Dungeon:8",
      "note": "Killed Grum",
      "turn": "10089",
      "type": "unique-killed",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "1",
        "skill": "Shields",
      },
      "level": "8",
      "location": "Dungeon:8",
      "note": "Reached skill level 1 in Shields",
      "turn": "10090",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "76",
        "level": "10",
        "mp": "10",
      },
      "level": "8",
      "location": "Dungeon:8",
      "note": "Reached XP level 10. HP: 45/76 MP: 10/10",
      "turn": "10111",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "5",
        "skill": "Shields",
      },
      "level": "8",
      "location": "Dungeon:8",
      "note": "Reached skill level 5 in Shields",
      "turn": "10510",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "bips": "*",
        "god": "Wu Jian",
      },
      "level": "8",
      "location": "Dungeon:8",
      "note": "Reached * piety under Wu Jian",
      "turn": "11233",
      "type": "piety-god",
    },
    {
      "branch": "Dungeon",
      "data": {
        "god": "Makhleb",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Found a burning altar of Makhleb.",
      "turn": "12916",
      "type": "altar",
    },
    {
      "branch": "Dungeon",
      "data": {
        "god": "Trog",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Found a bloodstained altar of Trog.",
      "turn": "12926",
      "type": "altar",
    },
    {
      "branch": "Dungeon",
      "data": {
        "god": "Elyvilon",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Found a white marble altar of Elyvilon.",
      "turn": "12926",
      "type": "altar",
    },
    {
      "branch": "Dungeon",
      "data": {
        "god": "Shining One",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Found a glowing golden altar of the Shining One.",
      "turn": "12927",
      "type": "altar",
    },
    {
      "branch": "Dungeon",
      "data": {
        "god": "Gozag",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Found an opulent altar of Gozag.",
      "turn": "12928",
      "type": "altar",
    },
    {
      "branch": "Dungeon",
      "data": {
        "god": "Uskayaw",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Found a hide-covered altar of Uskayaw.",
      "turn": "12928",
      "type": "altar",
    },
    {
      "branch": "Dungeon",
      "data": {
        "god": "Nemelex",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Found a sparkling altar of Nemelex Xobeh.",
      "turn": "12928",
      "type": "altar",
    },
    {
      "branch": "Dungeon",
      "data": {
        "item": "scroll of acquirement",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Identified a scroll of acquirement",
      "turn": "13319",
      "type": "ident",
    },
    {
      "branch": "Dungeon",
      "data": {
        "item": "scroll of acquirement",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Identified a scroll of acquirement",
      "turn": "13319",
      "type": "item",
    },
    {
      "branch": "Dungeon",
      "data": {
        "gold": "1070",
        "is_valid_shop": true,
        "item": "+18 shield of the Gong {rElec rN+ Will+ EV-5}",
      },
      "level": "7",
      "location": "Dungeon:7",
      "note": "Bought the +18 shield of the Gong {rElec rN+ Will+ EV-5} for 1070 gold pieces",
      "turn": "13443",
      "type": "bought",
    },
    {
      "branch": "Dungeon",
      "data": {
        "gold": "1070",
        "item": "+18 shield of the Gong {rElec rN+ Will+ EV-5}",
      },
      "level": "7",
      "location": "Dungeon:7",
      "note": "Bought the +18 shield of the Gong {rElec rN+ Will+ EV-5} for 1070 gold pieces",
      "turn": "13443",
      "type": "item",
    },
    {
      "branch": "Dungeon",
      "data": {
        "hp": "81",
        "level": "11",
        "mp": "11",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Reached XP level 11. HP: 81/81 MP: 11/11",
      "turn": "13772",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "level": "10",
        "skill": "Shields",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Reached skill level 10 in Shields",
      "turn": "13794",
      "type": "skill-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "bips": "**",
        "god": "Wu Jian",
      },
      "level": "9",
      "location": "Dungeon:9",
      "note": "Reached ** piety under Wu Jian",
      "turn": "13975",
      "type": "piety-god",
    },
    {
      "branch": "Orc",
      "data": {
        "who": "Urug",
      },
      "level": "1",
      "location": "Orc:1",
      "note": "Noticed Urug",
      "turn": "14788",
      "type": "unique-noticed",
    },
    {
      "branch": "Lair",
      "data": {
        "who": "Nergalle",
      },
      "level": "1",
      "location": "Lair:1",
      "note": "Noticed Nergalle",
      "turn": "14898",
      "type": "unique-noticed",
    },
    {
      "branch": "Lair",
      "data": {
        "who": "Nergalle",
      },
      "level": "1",
      "location": "Lair:1",
      "note": "Killed Nergalle",
      "turn": "14911",
      "type": "unique-killed",
    },
    {
      "branch": "Lair",
      "data": {
        "level": "14",
        "skill": "Dodging",
      },
      "level": "1",
      "location": "Lair:1",
      "note": "Reached skill level 14 in Dodging",
      "turn": "14914",
      "type": "skill-level",
    },
    {
      "branch": "Lair",
      "data": {
        "hp": "93",
        "level": "12",
        "mp": "12",
      },
      "level": "1",
      "location": "Lair:1",
      "note": "Reached XP level 12. HP: 93/93 MP: 12/12",
      "turn": "15236",
      "type": "experience-level",
    },
    {
      "branch": "Lair",
      "data": {
        "level": "15",
        "skill": "Dodging",
      },
      "level": "1",
      "location": "Lair:1",
      "note": "Reached skill level 15 in Dodging",
      "turn": "15985",
      "type": "skill-level",
    },
    {
      "branch": "Lair",
      "data": {
        "bips": "***",
        "god": "Wu Jian",
      },
      "level": "1",
      "location": "Lair:1",
      "note": "Reached *** piety under Wu Jian",
      "turn": "16050",
      "type": "piety-god",
    },
    {
      "branch": "Lair",
      "data": {
        "level": "1",
        "skill": "Short Blades",
      },
      "level": "1",
      "location": "Lair:1",
      "note": "Reached skill level 1 in Short Blades",
      "turn": "16113",
      "type": "skill-level",
    },
    {
      "branch": "Lair",
      "data": {
        "item": "+4 buckler of the Slorg {rC+++ rN+++ MP-9 Int-2}",
      },
      "level": "2",
      "location": "Lair:2",
      "note": "Identified the +4 buckler of the Slorg {rC+++ rN+++ MP-9 Int-2}",
      "turn": "16229",
      "type": "ident",
    },
    {
      "branch": "Lair",
      "data": {
        "item": "+4 buckler of the Slorg {rC+++ rN+++ MP-9 Int-2}",
      },
      "level": "2",
      "location": "Lair:2",
      "note": "Identified the +4 buckler of the Slorg {rC+++ rN+++ MP-9 Int-2}",
      "turn": "16229",
      "type": "item",
    },
    {
      "branch": "Lair",
      "data": {
        "level": "5",
        "skill": "Short Blades",
      },
      "level": "2",
      "location": "Lair:2",
      "note": "Reached skill level 5 in Short Blades",
      "turn": "16484",
      "type": "skill-level",
    },
    {
      "branch": "Lair",
      "data": {
        "bips": "****",
        "god": "Wu Jian",
      },
      "level": "2",
      "location": "Lair:2",
      "note": "Reached **** piety under Wu Jian",
      "turn": "17274",
      "type": "piety-god",
    },
    {
      "branch": "Lair",
      "data": {
        "level": "10",
        "skill": "Fighting",
      },
      "level": "2",
      "location": "Lair:2",
      "note": "Reached skill level 10 in Fighting",
      "turn": "17355",
      "type": "skill-level",
    },
    {
      "branch": "Lair",
      "data": {
        "level": "16",
        "skill": "Dodging",
      },
      "level": "3",
      "location": "Lair:3",
      "note": "Reached skill level 16 in Dodging",
      "turn": "17377",
      "type": "skill-level",
    },
    {
      "branch": "Lair",
      "data": {
        "hp": "103",
        "level": "13",
        "mp": "13",
      },
      "level": "3",
      "location": "Lair:3",
      "note": "Reached XP level 13. HP: 79/103 MP: 13/13",
      "turn": "17392",
      "type": "experience-level",
    },
    {
      "branch": "Lair",
      "data": {
        "level": "10",
        "skill": "Short Blades",
      },
      "level": "4",
      "location": "Lair:4",
      "note": "Reached skill level 10 in Short Blades",
      "turn": "18347",
      "type": "skill-level",
    },
    {
      "branch": "Lair",
      "data": {
        "bips": "*****",
        "god": "Wu Jian",
      },
      "level": "3",
      "location": "Lair:3",
      "note": "Reached ***** piety under Wu Jian",
      "turn": "18491",
      "type": "piety-god",
    },
    {
      "branch": "Lair",
      "data": {
        "hp": "113",
        "level": "14",
        "mp": "14",
      },
      "level": "4",
      "location": "Lair:4",
      "note": "Reached XP level 14. HP: 113/113 MP: 14/14",
      "turn": "19547",
      "type": "experience-level",
    },
    {
      "branch": "Lair",
      "data": {
        "level": "17",
        "skill": "Dodging",
      },
      "level": "4",
      "location": "Lair:4",
      "note": "Reached skill level 17 in Dodging",
      "turn": "19581",
      "type": "skill-level",
    },
    {
      "branch": "Lair",
      "data": {
        "item": "+12 flail of Duty {speed, rC+ MP+9 Int+5}",
      },
      "level": "5",
      "location": "Lair:5",
      "note": "Identified the +12 flail of Duty {speed, rC+ MP+9 Int+5}",
      "turn": "19734",
      "type": "ident",
    },
    {
      "branch": "Lair",
      "data": {
        "item": "+12 flail of Duty {speed, rC+ MP+9 Int+5}",
      },
      "level": "5",
      "location": "Lair:5",
      "note": "Identified the +12 flail of Duty {speed, rC+ MP+9 Int+5}",
      "turn": "19734",
      "type": "item",
    },
    {
      "branch": "Lair",
      "data": {
        "god": "Wu Jian",
      },
      "level": "5",
      "location": "Lair:5",
      "note": "Found an ornate altar of the Wu Jian Council.",
      "turn": "19755",
      "type": "altar",
    },
    {
      "branch": "Lair",
      "data": {
        "god": "Uskayaw",
      },
      "level": "5",
      "location": "Lair:5",
      "note": "Found a hide-covered altar of Uskayaw.",
      "turn": "19756",
      "type": "altar",
    },
    {
      "branch": "Lair",
      "data": {
        "item": "+1 ring mail of Uripeeb {Dex+4}",
      },
      "level": "5",
      "location": "Lair:5",
      "note": "Identified the +1 ring mail of Uripeeb {Dex+4}",
      "turn": "19871",
      "type": "ident",
    },
    {
      "branch": "Lair",
      "data": {
        "item": "+1 ring mail of Uripeeb {Dex+4}",
      },
      "level": "5",
      "location": "Lair:5",
      "note": "Identified the +1 ring mail of Uripeeb {Dex+4}",
      "turn": "19871",
      "type": "item",
    },
    {
      "branch": "Orc",
      "data": {
        "who": "Urug",
      },
      "level": "1",
      "location": "Orc:1",
      "note": "Killed Urug",
      "turn": "20609",
      "type": "unique-killed",
    },
    {
      "branch": "Orc",
      "data": {
        "name": "Cuatyt's General Store",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Found Cuatyt's General Store.",
      "turn": "21174",
      "type": "shop",
    },
    {
      "branch": "Orc",
      "data": {
        "level": "18",
        "skill": "Dodging",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Reached skill level 18 in Dodging",
      "turn": "21279",
      "type": "skill-level",
    },
    {
      "branch": "Orc",
      "data": {
        "name": "Atonkoc's General Store",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Found Atonkoc's General Store.",
      "turn": "21303",
      "type": "shop",
    },
    {
      "branch": "Orc",
      "data": {
        "who": "Donald",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Noticed Donald",
      "turn": "21382",
      "type": "unique-noticed",
    },
    {
      "branch": "Orc",
      "data": {
        "who": "Donald",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Killed Donald",
      "turn": "22225",
      "type": "unique-killed",
    },
    {
      "branch": "Orc",
      "data": {
        "level": "19",
        "skill": "Dodging",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Reached skill level 19 in Dodging",
      "turn": "22798",
      "type": "skill-level",
    },
    {
      "branch": "Orc",
      "data": {
        "name": "Zetzuo Viuh's Jewellery Boutique",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Found Zetzuo Viuh's Jewellery Boutique.",
      "turn": "22942",
      "type": "shop",
    },
    {
      "branch": "Orc",
      "data": {
        "name": "Lakez's Antique Armour Shop",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Found Lakez's Antique Armour Shop.",
      "turn": "22942",
      "type": "shop",
    },
    {
      "branch": "Orc",
      "data": {
        "branch": "Pandemonium",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Found a one-way gate leading to the halls of Pandemonium.",
      "turn": "22943",
      "type": "portal",
    },
    {
      "branch": "Orc",
      "data": {
        "gold": "148",
        "is_valid_shop": true,
        "item": "creamy pearl robe",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Bought a creamy pearl robe for 148 gold pieces",
      "turn": "22948",
      "type": "bought",
    },
    {
      "branch": "Orc",
      "data": {
        "gold": "148",
        "item": "+1 robe "Woud" {Regen+ Int+3}",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Identified the +1 robe "Woud" {Regen+ Int+3} (You bought it in a shop on level 2 of the Orcish Mines)",
      "turn": "22948",
      "type": "ident-bought-loc",
    },
    {
      "branch": "Orc",
      "data": {
        "gold": "148",
        "item": "+1 robe "Woud" {Regen+ Int+3}",
      },
      "level": "2",
      "location": "Orc:2",
      "note": "Identified the +1 robe "Woud" {Regen+ Int+3} (You bought it in a shop on level 2 of the Orcish Mines)",
      "turn": "22948",
      "type": "item",
    },
    {
      "branch": "Dungeon",
      "data": {
        "gold": "440",
        "is_valid_shop": true,
        "item": "tin of tremorstones (2/2)",
      },
      "level": "4",
      "location": "Dungeon:4",
      "note": "Bought a tin of tremorstones (2/2) for 440 gold pieces",
      "turn": "23186",
      "type": "bought",
    },
    {
      "branch": "Lair",
      "data": {
        "hp": "125",
        "level": "15",
        "mp": "15",
      },
      "level": "2",
      "location": "Lair:2",
      "note": "Reached XP level 15. HP: 87/125 MP: 15/15",
      "turn": "23532",
      "type": "experience-level",
    },
    {
      "branch": "Dungeon",
      "data": {
        "gold": "97",
        "is_valid_shop": true,
        "item": "scroll of enchant armour",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Bought a scroll of enchant armour for 97 gold pieces",
      "turn": "24446",
      "type": "bought",
    },
    {
      "branch": "Dungeon",
      "data": {
        "gold": "97",
        "is_valid_shop": true,
        "item": "scroll of enchant weapon",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Bought a scroll of enchant weapon for 97 gold pieces",
      "turn": "24446",
      "type": "bought",
    },
    {
      "branch": "Dungeon",
      "data": {
        "gold": "97",
        "is_valid_shop": true,
        "item": "scroll of enchant armour",
      },
      "level": "3",
      "location": "Dungeon:3",
      "note": "Bought a scroll of enchant armour for 97 gold pieces",
      "turn": "24446",
      "type": "bought",
    },
    {
      "branch": "Swamp",
      "data": {
        "god": "Gozag",
      },
      "level": "2",
      "location": "Swamp:2",
      "note": "Found an opulent altar of Gozag.",
      "turn": "24962",
      "type": "altar",
    },
    {
      "branch": "Swamp",
      "data": {
        "level": "20",
        "skill": "Dodging",
      },
      "level": "1",
      "location": "Swamp:1",
      "note": "Reached skill level 20 in Dodging",
      "turn": "25448",
      "type": "skill-level",
    },
    {
      "branch": "Swamp",
      "data": {
        "who": "Nikola",
      },
      "level": "2",
      "location": "Swamp:2",
      "note": "Noticed Nikola",
      "turn": "25621",
      "type": "unique-noticed",
    },
    {
      "branch": "Swamp",
      "level": "2",
      "location": "Swamp:2",
      "note": "Killed from afar by Nikola",
      "turn": "26211",
      "type": "last-event",
    },
  ],
  "fullVersion": "0.30-a0",
  "god": "Wu Jian",
  "isBloatcrawl": false,
  "isMorgue": true,
  "isSprint": false,
  "isTrunk": true,
  "is_bcrawl": false,
  "morgue": {
    "basename": "morgue-kareem-20220818-224217",
    "date": 2022-08-18T22:42:17.000Z,
    "filename": "morgue-kareem-20220818-224217.txt.gz",
    "player": "kareem",
    "timestamp": "1660862537000",
    "url": "https://crawl.kelbi.org/crawl/morgue/kareem/morgue-kareem-20220818-224217.txt.gz",
  },
  "runeCount": 0,
  "runes": [],
  "score": "72258",
  "species": "Merfolk",
  "stash": {
    "artefact_list": [],
    "branch": {},
    "shop_list": [],
  },
  "timeSeconds": 5002,
  "turns": 26211,
  "value": "13998099872284306217",
  "version": "0.30",
}
`;
